### PR TITLE
[Woo POS] M2: Products loading errors and empty state

### LINKED ISSUE #12001: [Woo POS] M2: Handle products loading errors, empty state
1. Products are empty and app fails to load them -> display global error as per designs with "Retry" botton
2. Failed to load next page -> automatic retry?

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #12001

### Description

* Adds empty and error state to the products list
* Hides cart for this states
* I do not pass errors from the splash screen as this seems to increase complexity for very small UX improvement (we may consider that to improve later)
* I changed the API of the ProductsSource class as I could not figure out how to properly implement the separation between full loading state and loading with pull to refresh (@samiuelson I think your advice will be really welcome here)

### Testing information

[Fixed_displayong_of_the_empty_state.patch](https://github.com/user-attachments/files/16360207/Fixed_displayong_of_the_empty_state.patch)
* Use the patch or your own imitations of the empty list and error
* Notice the empty and error states
* Overal products related (pagination, reloading) should be retested

### Images/gif

https://github.com/user-attachments/assets/4d1653d4-5db4-4e49-a3f4-577c67ffc868

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/products/WooPosProductsDataSource.kt [hunk 1]
```diff
@@ -18,21 +21,44 @@ class WooPosProductsDataSource @Inject constructor(
...
-        .map { it.filter { product -> product.price != null } }
-
-    suspend fun loadSimpleProducts(forceRefreshProducts: Boolean): Result<Unit> = withContext(Dispatchers.IO) {
+    fun loadSimpleProducts(forceRefreshProducts: Boolean): Flow<ProductsResult> = flow {
         if (forceRefreshProducts) {
             productStore.deleteProductsForSite(site.get())
         }
-        handler.loadFromCacheAndFetch(
+
+        val result = handler.loadFromCacheAndFetch(
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/products/WooPosProductsDataSource.kt:32]: @samiuelson As per the `loadFromCacheAndFetch` we don't use cached products at all?

```
        return if (searchQuery.isNotEmpty()) {
            when (searchType) {
                SearchType.DEFAULT -> {
                    searchInCache()
                    remoteSearch()
                }
                SearchType.SKU -> {
                    searchInCache()
                    remoteSearch()
                }
            }
        } else {
            fetchProducts()
        }
```

Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/products/WooPosProductsDataSource.kt:32]: I've created a ticket to figure this out - https://github.com/woocommerce/woocommerce-android/issues/12125

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/component/WooPosErrorState.kt [hunk 2]
```diff
@@ -16,14 +17,17 @@ import androidx.compose.ui.Alignment
...
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import com.woocommerce.android.R
 import com.woocommerce.android.ui.woopos.common.composeui.WooPosPreview
+import com.woocommerce.android.ui.woopos.common.composeui.WooPosTheme
+import com.woocommerce.android.ui.woopos.common.composeui.toAdaptivePadding
 
 @Composable
 fun WooPosErrorState(
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/component/WooPosErrorState.kt:28]: @backwardstruck jfyi - I updated the UI of the error state as per designs

Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/component/WooPosErrorState.kt:28]: Got it. Thanks for letting me know. I'll merge this with the changes I made in this PR:
https://github.com/woocommerce/woocommerce-android/pull/12133

Will wait for your PR to merge first so I can resolve any conflicts.

**Comment by backwardstruck** [comment]
Tested with and without the patch. Saw the empty and error states.
Tested pagination and reloading.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/splash/WooPosSplashScreen.kt [hunk 3]
```diff
@@ -36,7 +36,6 @@ fun WooPosSplashScreen(onNavigationEvent: (WooPosNavigationEvent) -> Unit) {
     when (state.value) {
         is WooPosSplashState.Loading -> Loading()
         is WooPosSplashState.Loaded -> onNavigationEvent(WooPosNavigationEvent.OpenHomeFromSplash)
-        WooPosSplashState.Error -> TODO()
```
Comment (backwardstruck) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/splash/WooPosSplashScreen.kt:39]: 👍

**Review by backwardstruck — APPROVED** [review]

**Comment by staskus** [comment]
@kidinov, @backwardstruck  noticed a typo on "Retry"

image

**Comment by kidinov** [comment]
@staskus, thanks for noticing! I'll fix in the following PR 🙇

**Pull request merged**